rust/windows-client/src-tauri/src/client/gui.rs

```diff
@@ -320,14 +354,15 @@ impl connlib_client_shared::Callbacks for CallbackHandler {
 struct Controller {
     /// Debugging-only settings like API URL, auth URL, log filter
     advanced_settings: AdvancedSettings,
-    /// mpsc sender to send things to the controller task
-    ctlr_tx: CtlrTx,
     /// connlib / tunnel session
     connlib_session: Option<connlib_client_shared::Session<CallbackHandler>>,
     /// The UUIDv4 device ID persisted to disk
     /// Sent verbatim to Session::connect
     device_id: String,
     logging_handles: client::logging::Handles,
+    // TODO: Use `select!` over two notifiers instead of funneling everything through just one
+    notify_controller: Arc<Notify>,
+    resources: Arc<ArcSwap<Vec<ResourceDescription>>>,
```

Comment:
If we already have a resource list in `CallbackHandler` why keep a resource list here?

Reply:
This one I think is for the clipboard copy-paste, it associates the ID that the system tray callback sends us with the pastable domain / CIDR, so that I don't have to think about passing emojis or non-Latin1 chars through the system tray code.

And I pass the handler to connlib without an Arc, so I lose track of it

Reply:
You could store a clone of the `CallbackHandler` and retrieve the resources from there, as it's the relation between the notify and resources is not obvious.

Reply:
Oh yeah, cause it implements Clone. I'll clean this up today

Reply:
Fixed in c6b118800